OpenSCAP/scap-security-guide
# Script to auto-upload / update ansible galaxy roles from SSG

#### Description:

- This is a resubmit of #2466 with substantial changes:
- The upload no longer happens as part of build, instead there is a script in misc that does all the updating
- The script now takes tokens as well as plain text logins, tokens are heavily recommended over login via username
- Be exception safe, if something fails don't leave the cwd at the wrong location
- A ton of minor readability fixes

Keep in mind that this script is still not perfect. Despite my efforts to make it robust it will still produce undefined states especially when creating a repo is successful but uploading the initial content is not. If that happens the next time it is run it will fail. There are numerous other issues like this. It is very hard for me to fix them all because of how the script is written.

How to test:
```
cd build/
cmake -G Ninja ..
ninja
../shared/misc/upload_ansible_roles_to_galaxy.py --build-roles-dir ./roles/ --meta-template-path ../shared/misc/galaxy_ansible_role_meta_template.yml

Input your GitHub credentials:
username or token: REDACTED
password (or empty for token): 
Creating new Github repository: ssg-centos7-role-stig-satellite-upstream
Creating new Github repository: ssg-rhel7-role-stig-satellite-upstream
Creating new Github repository: ssg-sl7-role-stig-satellite-upstream
Cloning into 'ssg-centos7-role-stig-satellite-upstream'...
warning: You appear to have cloned an empty repository.
- ssg-centos7-role-stig-satellite-upstream was created successfully
[master (root-commit) 1675cfb] Initial commit
 8 files changed, 110 insertions(+)
 create mode 100644 README.md
 create mode 100644 defaults/main.yml
 create mode 100644 handlers/main.yml
 create mode 100644 meta/main.yml
 create mode 100644 tasks/main.yml
 create mode 100644 tests/inventory
 create mode 100644 tests/test.yml
 create mode 100644 vars/main.yml
Counting objects: 16, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (10/10), done.
Writing objects: 100% (16/16), 2.47 KiB | 843.00 KiB/s, done.
Total 16 (delta 0), reused 0 (delta 0)
To github.com:OpenSCAP-Ansible-Roles/ssg-centos7-role-stig-satellite-upstream.git
 * [new branch]      master -> master
Cloning into 'ssg-rhel7-role-stig-satellite-upstream'...
warning: You appear to have cloned an empty repository.
- ssg-rhel7-role-stig-satellite-upstream was created successfully
[master (root-commit) 402a660] Initial commit
 8 files changed, 110 insertions(+)
 create mode 100644 README.md
 create mode 100644 defaults/main.yml
 create mode 100644 handlers/main.yml
 create mode 100644 meta/main.yml
 create mode 100644 tasks/main.yml
 create mode 100644 tests/inventory
 create mode 100644 tests/test.yml
 create mode 100644 vars/main.yml
Counting objects: 16, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (10/10), done.
Writing objects: 100% (16/16), 2.46 KiB | 2.46 MiB/s, done.
Total 16 (delta 0), reused 0 (delta 0)
To github.com:OpenSCAP-Ansible-Roles/ssg-rhel7-role-stig-satellite-upstream.git
 * [new branch]      master -> master
Cloning into 'ssg-sl7-role-stig-satellite-upstream'...
warning: You appear to have cloned an empty repository.
- ssg-sl7-role-stig-satellite-upstream was created successfully
[master (root-commit) b3d3409] Initial commit
 8 files changed, 110 insertions(+)
 create mode 100644 README.md
 create mode 100644 defaults/main.yml
 create mode 100644 handlers/main.yml
 create mode 100644 meta/main.yml
 create mode 100644 tasks/main.yml
 create mode 100644 tests/inventory
 create mode 100644 tests/test.yml
 create mode 100644 vars/main.yml
Counting objects: 16, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (10/10), done.
Writing objects: 100% (16/16), 2.46 KiB | 1.23 MiB/s, done.
Total 16 (delta 0), reused 0 (delta 0)
To github.com:OpenSCAP-Ansible-Roles/ssg-sl7-role-stig-satellite-upstream.git
 * [new branch]      master -> master
Processing ssg-centos6-role-C2S...
Processing ssg-centos6-role-CS2...
Processing ssg-centos6-role-CSCF-RHEL6-MLS...
Processing ssg-centos6-role-common...
Processing ssg-centos6-role-desktop...
...
```

#### Rationale:

- Manually updating ansible roles is very tedious and nobody can be expected to update ~130 ansible roles every SSG release. This script should make that easier
